### ISSUE TITLE
Native representation of char type

### ISSUE THREAD

**Issue by imkiva** [covscript/drafts#1]
Our parser parse all input source as `std::u32string`, whose underlying char type is `char32_t`.
It's natural to use `char32_t` as char literal type.

from cppreference.com:
> char32_t is an unsigned integer type used for 32-bit wide characters and is the same type as uint_least32_t
we can implement char type as int in the virtual machine.

**Comment by MouriNaruto** [comment]
Looks like Python, why not UTF-8? (UTF-8 is more compact.)

**Comment by imkiva** [comment]
> Looks like Python, why not UTF-8? (UTF-8 is more compact.)
@MouriNaruto 

Yes, it is utf-8, at least in users' eyes, and we only support utf-8. I think it's much simpler to use `char32_t` (or `int32_t`) to store the code point inside the VM (in fact, the VM only has three types: `int`, `double` and `any`) and during the lexer process, because a `char32_t` is large enough to store a single Chinese character, as a result, our `string` type can finally do things right such as `split`, `find`, `length`, etc.

But you are right, using a 32-bit integer to store a char is a little wasteful....
and moreover, we may need an 8-bit `byte` type for IO purpose (?)

I am not sure if there's better solutions. Any suggestions are all welcomed :)

**Comment by ice1000** [comment]
Maybe we can use a type like Haskell's `Text` and provide conversion against a lazy stream of utf8 characters?

**Comment by mikecovlee** [comment]
#2, but use u8char in covscript.

**Comment by mikecovlee** [comment]
We will provide int8_t, uint8_t, int32_t, uint32_t, int64_t, uint64_t in virtual machine.

**Comment by imkiva** [comment]
> #2, but use u8char in covscript.

Can you explain in details?
